### PR TITLE
Add reviewdog

### PR DESCRIPTION
**Pull request by litleleprikon**

### CODE REVIEW

File: tmp.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package moira
+
+type tmp struct {
```
Comment (litleleprikon) [inline tmp.go]: **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
type `tmp` is unused (unused)

File: tmp.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+	field1 bool
+	field2 string
+	field3 bool
+	field4 int
+	field5 bool
```
Comment (litleleprikon) [inline tmp.go]: **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
`field5` is unused (structcheck)

File: tmp.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package moira
+
+type tmp struct {
+	field1 bool
```
Comment (litleleprikon) [inline tmp.go]: **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
`field1` is unused (structcheck)

File: tmp.go [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+	field2 string
```
Comment (litleleprikon) [inline tmp.go]: **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
`field2` is unused (structcheck)

File: tmp.go [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+	field3 bool
```
Comment (litleleprikon) [inline tmp.go]: **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
`field3` is unused (structcheck)

File: tmp.go [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+	field4 int
```
Comment (litleleprikon) [inline tmp.go]: **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
`field4` is unused (structcheck)

**Comment by litleleprikon** [comment]
@beevee This is how review dog works. I will remove temporary commit after you will check the result of review.
I also added the `diff` option that checks only changed files comparing with master. I believe that if we will lint only changed files we can add all linters and step by step increate the quality of our code base.

**Comment by beevee** [comment]
I don't like that it leaves comments from your account. Can we make it work from a service account?

Also, please show what it looks like when author force-pushes different code.

File: tmp.go [hunk 7]
```diff
@@ -0,0 +1,6 @@
+package moira
+
+type tmp struct {
```
Comment (litleleprikon) [inline tmp.go]: **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
`tmp` is unused (deadcode)

**Comment by beevee** [comment]
Proposed behavior:

1. Reviewdog check becomes required to merge PRs.
2. Travis CI check does *not* fail on linter errors.

File: index/search.go [hunk 8]
```diff
@@ -23,7 +23,7 @@ func (index *Index) checkIfIndexIsReady() bool {
 	if index.IsReady() {
 		return true
 	}
-	timeout := time.After(indexWaitTimeout)
```
Comment (beevee) [inline index/search.go]: This change avoids linter detection. Reviewdog only checks current PR diff for linter errors, but this change creates an error elsewhere (makes a constant unused). Previous linter would have complained.

**Comment by beevee** [comment]
Haha, seems working!

**Comment by beevee** [comment]
Still haven't figured out how to make local reporter work without diff.

**Comment by beevee** [comment]
Had to make this simpler, because reviewdog doesn't work without diff command locally.

So, I've split old-style `lint` and new `reviewdog` into separate makefile sections. We'll run `make lint` locally and `make reviewdog` on CI. This is not ideal, because actual linter commands used with and without reviewdog may diverge over time, but we still have a single `golangci.yml` to control both.

I've removed the reporter configuration variable, since I couldn't make it work for any reporter other than `github-check`.

Still have a problem, though. There are usually two builds for our PRs: one for branch and one for PR. They can have different lint results. But reviewdog posts check result for both builds under the same name, so they overwrite each other in unpredictable order. I'm almost certain that I can construct a proof-of-concept for missing lint warnings. Again.

**Comment by litleleprikon** [comment]
Looks ok. I cannot approve this PR by myself as I am the creator of it.

**Pull request merged**